### ISSUE TITLE
Component's ngOnInit called twice when using componentOutlet 

### ISSUE THREAD

**Issue by mrkosima** [laco0416/angular2-component-outlet#9]
When component using `componentOutlet` directive, `ngOnInit` called twice.
For components without this directive `ngOnInit` called once.

Please, look into console
https://embed.plnkr.co/vTTRaWC4oOuU5TSZxLfk/

**Comment by lacolaco** [comment]
Thank you for your report.
The second `ngOnInit` is called in `ComponentOutlet`.
https://plnkr.co/edit/Ixo8Nni8X7dHguv3i0q9?p=preview

Current `ComponentOutlet` inherits all members including `ngOnInit` from the `context`. This behavior is not good, but I think... this is not a bug.

example use-case

``` ts
@Component({
  selector: 'dynamic-component',
  template: `
    <span *componentOutlet="template; context: context; selector:'dynamic-component'"></span>
  `,
})
export class DynamicComponent {
  context = {
    timer: 0,
    ngOnInit: function() {
      setInterval(() => {this.timer++}, 1000);
    }
  };
  template: string = '<p>{{timer}}</p>';
}
```

You can pass `ngOnInit` into the dynamic component. This is just a feature.

**Comment by lacolaco** [comment]
It seems solved in v3.0.0

https://plnkr.co/edit/nwzYXK3hRlAq9lg0RqCa?p=preview

**Comment by riteshgupta5555** [comment]
plunkr does not work
